pkg/util/awsutil/s3factory/client.go

```diff
@@ -88,7 +91,7 @@ func setupAWSConfig(kubecli kubernetes.Interface, ns, secret, configDir string)
 	}
 
 	config := se.Data[api.AWSSecretConfigFileName]
-	if len(config) != 0 {
+	if len(config) > 0 {
```

Comment:
i think original code works as expected. no need to change. I want to keep pr to be minimum as possible.